src/share/poudriere/include/fs.sh

```diff
@@ -23,6 +23,15 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
+remove_altroot() {
+	[ $# -ne 1 ] && eargs remove_altroot mountpoint
+	local mountpoint altroot
+	mountpoint="$1"
+	altroot="$([ -z "${NO_ZFS}" ] && zpool get -Ho value altroot "${ZPOOL}")"
```

Comment:
Why the `$()` for this? We don't need a subshell for the test; we may avoid a fork if `NO_ZFS` is set.
We need to set a default value or unset `altroot` too for `NO_ZFS`.
```
altroot=
if [ -z "${NO_ZFS}" ]; then
  altroot=$(zpool get -Ho value altroot "${ZPOOL}")
fi
```
Or
```
altroot=
[ -n "${NO_ZFS}" ] || altroot=$(zpool get -Ho value altroot "${ZPOOL}")
```

The `test && test` will exit with `set -e`, even if in a subshell. Using `||` won't exit.
```
# sh -c 'set -e; foo=$(false && true); echo here'; echo $?
1
# sh -c 'set -e; foo=$(false && true) || :; echo here'; echo $?
here
0
# sh -c 'set -e; foo=$(false || true); echo here'; echo $?
here
0
```

Reply:
Done. Thanks for the detailed example. My shell-fu is not very strong :)